### PR TITLE
feat: cache

### PR DESCRIPTION
**Pull request by SonyPradana**
| Q            | A                                                         |
|--------------|-----------------------------------------------------------|
| Is bugfix?   | **No**                                              |
| New feature? | **Yes**                                              |
| Breaks BC?   | **No**                                              |
| Fixed issues | |

------
Cache manager

### CODE REVIEW

File: split-repo.php [hunk 1]
```diff
@@ -5,6 +5,7 @@
 return [
     'tag_version' => "0.35.6",
     'split_repositorys' => [
+        'sonypradana/Cache' => '/src/System/Cache/',
```
Comment (SonyPradana) [inline split-repo.php:8]: use lowercase

File: src/System/Cache/CacheInterface.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+     * @param iterable<string> $keys a list of keys that can be obtained in a single operation
+     *
+     * @return iterable<string, mixed> A list of key => value pairs. Cache keys that do not exist or are stale will have $default as value.
+     */
+    public function getMultiple(iterable $keys, mixed $default = null): iterable;
+
+    /**
+     * Persists a set of key => value pairs in the cache, with an optional TTL.
+     */
+    public function setMultiple(iterable $values, int|\DateInterval|null $ttl = null);
```
Comment (SonyPradana) [inline src/System/Cache/CacheInterface.php:47]: forgot return type

File: src/System/Cache/CacheManager.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+        return $this->default_driver;
+    }
+
+    /**
+     * @param array<int, mixed> $parameters
+     */
+    public function __call(string $method, array $parameters): mixed
+    {
+        return $this->driver()->{$method}($parameters);
```
Comment (SonyPradana) [inline src/System/Cache/CacheManager.php:49]: use parameter destruction

**Review by SonyPradana — COMMENTED** [review]
consider pass php stan before continue (write test)

**Comment by SonyPradana** [comment]
add
- [ ] increment and decrement
- [ ] remember
- [ ] mtime / meta data (make time)

File: src/System/Support/Facades/Cache.php [hunk 4]
```diff
@@ -0,0 +1,28 @@
+<?php
```
Comment (anggerpradana) [inline src/System/Support/Facades/Cache.php:1]: more doc

**Review by anggerpradana — CHANGES_REQUESTED** [review]

**Pull request merged**